src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -670,9 +632,7 @@ public override Assembly GetSatelliteAssembly(CultureInfo culture, Version versi
             an.CultureInfo = culture;
             an.Name = GetSimpleName() + ".resources";
 
-            RuntimeAssembly retAssembly = nLoad(an, null, this, ref stackMark,
-                                IntPtr.Zero,
-                                throwOnFileNotFound);
+            RuntimeAssembly retAssembly = nLoad(an, null, ref stackMark, throwOnFileNotFound, IntPtr.Zero);
```

Comment:
Why is it ok to not pass the requesting assembly here anymore?

Reply:
I replaced it with https://github.com/dotnet/coreclr/pull/22716/files#diff-656cc6f286474bdd1af492899ad62123R621 instead

Reply:
`StackCrawlMark.LookForMe` in CoreLib will return CoreLib once resolved. So the requesting assembly is always going be CoreLib with this change. Different behavior.

Reply:
I think that's same behaviour, before there was `.., this, ref stackMark, ..` where `stackMark` was unused and `this` was used. Now there is only 1 argument which is always this-like

Reply:
I don't think it's the same behavior. You are trying to resolve a satellite assembly for some other assembly that comes from ResourceManager. For example, resolving to "<AppDirectory>\cs\Boo.resources.dll" when `this` is assembly representing `<AppDirectory>\Boo.dll`.

Reply:
oh you are right, will update the PR